src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -664,9 +676,17 @@ private void verifySlotExpanded(WebElement element) {
   }
 
   private void verifyNoAds() {
-    Collection<String> slotsSelectors = AdsContent.getAllSlotsSelectors();
-    for (String selector : slotsSelectors) {
-      verifyNoAd(selector);
+    Map<String, String> slots = AdsContent.getSlotsSelectorsMap();
+    for (Map.Entry<String, String> entry : slots.entrySet()) {
+      final String slotName = entry.getKey();
+      final String selector = entry.getValue();
+      final String javaScriptTrigger = AdsContent.getSlotTrigger(slotName);
+
+      if (StringUtils.isEmpty(javaScriptTrigger)) {
+        verifyNoAd(selector);
+      } else {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)